### ISSUE TITLE
False positive for CA1508 on IsNullOrWhiteSpace when IndexOf is used afterwards

### ISSUE THREAD

**Issue by jnm2** [dotnet/roslyn-analyzers#5008]
### Analyzer

**Diagnostic ID**: [CA1508](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca1508)

### Analyzer source

**SDK**: [Built-in CA analyzers in .NET 5 SDK or later](https://docs.microsoft.com/dotnet/fundamentals/productivity/code-analysis)

**Version**: [SDK 5.0.201](https://dotnet.microsoft.com/download/dotnet/5.0)

```cs
using System;

while (Console.ReadLine() is { } line)
{
    // ⚠ CA1508 'string.IsNullOrWhiteSpace(line)' is always 'true'. Remove or refactor the condition(s) to avoid dead code.
    //  ↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓↓
    if (string.IsNullOrWhiteSpace(line)) continue;

    Console.WriteLine(line.IndexOf('1', StringComparison.Ordinal));
}
```

```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>net5.0</TargetFramework>
    <AnalysisMode>AllEnabledByDefault</AnalysisMode>
  </PropertyGroup>

</Project>
```

**Comment by mavasani** [comment]
I believe this is fixed in the latest NuGet package of `Microsoft.CodeAnalysis.NetAnalyzers` version `5.0.3` and also in .NET 6 branch of the analyzers

**Comment by mavasani** [comment]
This is fixed now.
